splunk/src/main/java/com/splunk/HttpService.java

```diff
@@ -193,6 +208,21 @@ public String getHost() {
     public int getPort() {
         return this.port;
     }
+    
+    /**
+     * Sets Custom Headers of this service
+     * 
+     * @param headers
+     */
+    public void setCustomHeaders(Map<String, String> headers) {
+    	if (Objects.nonNull(headers) && !headers.isEmpty()) {
+    		Map<String, String> fitleredCustomHeaders = headers.entrySet()

```

Comment:
I think we can remove this logic and the `filtertHttpHeaderKeys` above since the custom header values are only added if the header does not already exist down on line 486:
```java
if (!header.containsKey(key)) {
    cn.setRequestProperty(key, entry.getValue());
}
```

Reply:
Yes, sure that makes sense. Also, I was thinking about whether the customHeaders map needs to be cleared out before populating it if it's not empty?

Reply:
Oh good call that makes, although if we're not filtering I think a simple reassignment would suffice? e.g.
```java
if (Objects.nonNull(headers)) {
    customHeaders = headers;
}
```
What do you think?

Reply:
That would cover a case where someone needed to clear the customHeaders that were already set (I can't think of a use case for that, but potentially useful)

Reply:
Yes, I think a reassignment should suffice. I have made the necessary changes. Please check.